### PR TITLE
chore(web): add location field

### PR DESCRIPTION
**Pull request by nina992**
# Overview
 Add Location field 
## What I've done
image

### CODE REVIEW

File: web/src/beta/components/fields/PropertyFields/index.tsx [hunk 1]
```diff
@@ -61,6 +63,14 @@ const PropertyFields: React.FC<Props> = ({ propertyId, item }) => {
             description={sf.description}
             onChange={handlePropertyValueUpdate(item.schemaGroup, propertyId, sf.id, sf.type)}
           />
+        ) : sf.type == "latlng" ? (
```
Comment (KaWaite) [inline web/src/beta/components/fields/PropertyFields/index.tsx:66]: triple equals please

File: web/src/beta/components/fields/LocationField/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export default meta;
+
+type Story = StoryObj<typeof LocationField>;
+
+export const LocationFieldUI: Story = {
+  render: () => (
+    <LocationField name="Location" description="The Location field" onChange={action("onchange")} />
+  ),
+};
```
Comment (jashanbhullar) [inline web/src/beta/components/fields/LocationField/index.stories.tsx:18]:
```suggestion
export const Sample: Story = {
  args: {
    name: "Location Field",
    description: "Location field description",
    value: { lat: 87.43, lng: 107.53 },
    onChange: () => console.log("clicked"),
  },
};
```

If you write it like this, the props can be changed in the stories UI

File: web/src/beta/components/fields/LocationField/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    <Property name={name} description={description}>
+      <Wrapper>
+        <NumberInput
+          inputDescription="Latitude"
+          onChange={newValue => handleChange("Latitude", newValue)}
+        />
+        <NumberInput
+          inputDescription="Longtitude"
+          onChange={newValue => handleChange("Longtitude", newValue)}
+        />
```
Comment (jashanbhullar) [inline web/src/beta/components/fields/LocationField/index.tsx:45]:
```suggestion
        <NumberInput 
         value={location.lat}
          inputDescription="Latitude"
          onChange={newValue => handleChange("Latitude", newValue)}
        />
        <NumberInput 
        value={location.lng}
          inputDescription="Longtitude"
          onChange={newValue => handleChange("Longtitude", newValue)}
        />
```

Values should be passed down right?

Comment (jashanbhullar) [reply web/src/beta/components/fields/LocationField/index.tsx:45]: Also, font-size is 12px!

File: web/src/beta/components/fields/LocationField/index.stories.tsx (hunk 2 above)
Comment (jashanbhullar) [reply web/src/beta/components/fields/LocationField/index.stories.tsx:18]: image

File: web/src/beta/components/fields/LocationField/index.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { LatLng } from "@reearth/beta/utils/value";
+import { styled } from "@reearth/services/theme";
+
+import Property from "..";
+import NumberInput from "../common/NumberInput";
+
+type Props = {
+  name?: string;
```
Comment (jashanbhullar) [inline web/src/beta/components/fields/LocationField/index.tsx:10]: Should there be a disabled state for the field?

**Review by jashanbhullar — CHANGES_REQUESTED** [review]

File: web/src/beta/components/fields/PropertyFields/index.tsx [hunk 5]
```diff
@@ -1,4 +1,6 @@
 import TextInput from "@reearth/beta/components/fields/TextInput";
+import { LatLng } from "@reearth/beta/utils/value";
+import LocationField from "@reearth/classic/components/molecules/EarthEditor/PropertyPane/PropertyField/LocationField";
```
Comment (KaWaite) [inline web/src/beta/components/fields/PropertyFields/index.tsx:3]: You are using classic's component... :P

File: web/src/beta/components/fields/LocationField/index.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    </Property>
+  );
+};
+
+export default TextInput;
+
+const Wrapper = styled.div`
+  display: flex;
+  align-items: flex-start;
+  width: 289px;
```
Comment (KaWaite) [inline web/src/beta/components/fields/LocationField/index.tsx:56]: you shouldn't set a width. It should flex the same as the design. Please play with the design to see how it flexes

File: web/src/beta/components/fields/LocationField/index.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    if (newValue === undefined) return;
+
+    setLocation(prevLocation => ({
+      ...prevLocation,
+      [coordination === "Latitude" ? "lat" : "lng"]: newValue,
+    }));
+  }, []);
+  useEffect(() => {
+    if (location) onChange?.(location);
+  }, [location, onChange]);
```
Comment (KaWaite) [inline web/src/beta/components/fields/LocationField/index.tsx:33]: Will this not produce an infinite loop?

File: web/src/beta/components/fields/LocationField/index.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import NumberInput from "../common/NumberInput";
+
+type Props = {
+  name?: string;
+  description?: string;
+  value?: LatLng;
+  onChange?: (location: LatLng) => void;
+};
+
+const TextInput: React.FC<Props> = ({ name, description, value, onChange }) => {
```
Comment (KaWaite) [inline web/src/beta/components/fields/LocationField/index.tsx:16]: Name is wrong

File: web/src/beta/components/fields/PropertyFields/index.tsx (hunk 1 above)
Comment (nina992) [reply web/src/beta/components/fields/PropertyFields/index.tsx:66]: done

File: web/src/beta/components/fields/LocationField/index.tsx (hunk 4 above)
Comment (nina992) [reply web/src/beta/components/fields/LocationField/index.tsx:10]: I don't think so it is not exist in design

File: web/src/beta/components/fields/PropertyFields/index.tsx (hunk 5 above)
Comment (nina992) [reply web/src/beta/components/fields/PropertyFields/index.tsx:3]: sorry by mistake, fixed

File: web/src/beta/components/fields/LocationField/index.tsx (hunk 8 above)
Comment (nina992) [reply web/src/beta/components/fields/LocationField/index.tsx:16]: sure , fixed

File: web/src/beta/components/fields/LocationField/index.tsx (hunk 6 above)
Comment (nina992) [reply web/src/beta/components/fields/LocationField/index.tsx:58]: sure, removed

File: web/src/beta/components/fields/LocationField/index.tsx (hunk 7 above)
Comment (nina992) [reply web/src/beta/components/fields/LocationField/index.tsx:33]: I don't think so, can you explain more why ?

Comment (nina992) [reply web/src/beta/components/fields/LocationField/index.tsx]: fixed

File: web/src/beta/components/fields/PropertyFields/index.tsx [hunk 9]
```diff
@@ -1,4 +1,6 @@
+import LocationField from "@reearth/beta/components/fields/LocationField";
 import TextInput from "@reearth/beta/components/fields/TextInput";
+import { LatLng } from "@reearth/beta/utils/value";
```
Comment (KaWaite) [inline web/src/beta/components/fields/PropertyFields/index.tsx:3]:
```suggestion
import { type LatLng } from "@reearth/beta/utils/value";
```

**Review by KaWaite — APPROVED** [review]

**Review by jashanbhullar — APPROVED** [review]

**Pull request merged**